### ISSUE TITLE
flood_fill and flood are both in segmentation and morphology

### ISSUE THREAD

**Issue by emmanuelle** [scikit-image/scikit-image#4618]
The functions `flood_fill` and `flood` are both in the segmentation and in the morphology submodules, and neither is deprecated. I don't think we should keep the two versions in the API, but that we should deprecate one of them (probably keep segmentation?).

@JDWarner

**Comment by rfezzani** [comment]
This reminds me this discussion https://github.com/scikit-image/scikit-image/pull/4400#issuecomment-577231251 ;-)

**Comment by soupault** [comment]
I remember these two original arguments by @JDWarner - https://github.com/scikit-image/scikit-image/pull/3245#issuecomment-405047996, https://github.com/scikit-image/scikit-image/pull/3245#issuecomment-455918092. To me, both of them still sound reasonable.

**Comment by jni** [comment]
Yeah, this is the recurring problem of tags vs folders. For a while, we decided to treat our modules as tags (watershed was also in both morphology and segmentation), as a solution to the problem that even advanced users such as myself were unable to find said watershed (after my proposal of a flat namespace was rejected. ;)

I think our latest consensus is that we want our modules to reflect *purpose* rather than the discipline from which they stem. So I think flood should probably go in segmentation, if we pick one.
